### PR TITLE
First round of template simplification: separate generation of data files from actual creation

### PR DESCRIPTION
**Pull request by RobertKrawitz**
Generate the benchmark and pod YAML files separately from writing them out.  The goal here is to be able to generate golden files without having to extract them from the current_run directories.  This also avoids generation of some temporary files that have to be cleaned up when they are only used for generating intermediate YAML that's simply a dictionary for creating payload YAML files.

Still want to clean up all uses of update_environment_variable and delete_generate_file to use the new APIs in update_data_template_yaml_with_environment_variables.py and remove those functions.

### CODE REVIEW

**Comment by RobertKrawitz** [comment]
Golden file comparison passed with latest rebase.

File: benchmark_runner/benchmark_operator/workload_flavors/generate_yaml_from_workload_flavors.py [hunk 1]
```diff
@@ -40,112 +61,94 @@ def __get_yaml_template_by_workload(self, workload: str, extension='.yaml', skip
...
         """
-        # replace environment variables and generate hammerdb_data.yaml
-        if 'kata' in workload:
-            environment_variables.environment_variables_dict['kind'] = 'kata'
+        if '_pod' in workload:
+            return 'pod', 'pod'
+        elif '_kata' in workload:
+            return 'kata', 'pod'
+        elif '_vm' in workload:
+            return 'vm', 'vm'
```
Comment (ebattat) [inline benchmark_runner/benchmark_operator/workload_flavors/generate_yaml_from_workload_flavors.py:78]: Why we need to return couples ?

**Review by ebattat — COMMENTED** [review]
Can you please rebase it and rerun pytest against Func env ?

**Pull request merged**